Watchman.Configuration/Generic/AutoScalingResourceConfig.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+
+namespace Watchman.Configuration.Generic
+{
+    public class AutoScalingResourceConfig : IServiceAlarmConfig<AutoScalingResourceConfig>
```

Comment:
This is the "Curiously Recurring Template Pattern" ? http://zpbappi.com/curiously-recurring-template-pattern-in-csharp/
Is it necessary?

Reply:
yes, in `ServiceAlarmBuilder` we need to merge the configuration defined at service level with that at resource level. And the return type of the merge has to be the type representing the custom config.